src/events/eventToAction.js

```diff
@@ -292,6 +292,10 @@ export default (state: GlobalState, event: $FlowFixMe): EventAction | null => {
         // stop sending `operation`; see #4238.
         op: event.op ?? event.operation,
 
+        message_info_dict: event.message_info_dict
+          ? new Map(Object.entries(event.message_info_dict).map(([k, v]) => [parseInt(k, 10), v]))
+          : undefined,
+
```

Comment:
Probably cleanest to push this conversion down into the one reducer that consumes this action, so that the action type doesn't need to diverge from the event (more than it does already).

(We discussed this when you asked about it in person; mentioning here FTR in the thread.)